### PR TITLE
tools: ecds server

### PR DESCRIPTION
**Pull request by kyessenov**
Signed-off-by: Kuat Yessenov <kuat@google.com>

**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

**Comment by kyessenov** [comment]
This is currently blocked on 1.7 due to https://github.com/istio/istio/pull/26590 and istiod blocking `config_discovery` field. It works fine in isolation, or on master.

**Review by kyessenov — COMMENTED** [review]
@bianpengyuan @mandarjog This is ready and tested against 1.7.1.

File: tools/extensionserver/convert.go
Comment (mandarjog) [inline tools/extensionserver/convert.go:62]: Is this type of code due to one-off?

Comment (kyessenov) [reply tools/extensionserver/convert.go:62]: Yes.

**Comment by kyessenov** [comment]
Gentle ping.

File: tools/extensionserver/main/main.go
Comment (bianpengyuan) [inline tools/extensionserver/main/main.go:72]: Maybe worth also logging successful deletion event here.

**Review by bianpengyuan — APPROVED** [review]
LG, only some nits. Please unblock after resolving.

File: test/envoye2e/inventory.go
Comment (kyessenov) [reply test/envoye2e/inventory.go:61]: This is a bug in the test framework. V8 compiled shows up as v8#01.

File: test/envoye2e/stats_plugin/stats_test.go
Comment (kyessenov) [reply test/envoye2e/stats_plugin/stats_test.go:392]: Ack.

File: tools/extensionserver/main/main.go
Comment (kyessenov) [reply tools/extensionserver/main/main.go:72]: Done

**Comment by kyessenov** [comment]
/test test-asan_proxy

(timeout)

**Pull request merged**
